Source/WebCore/rendering/mac/RenderThemeMac.mm

```diff
@@ -465,9 +465,9 @@ static Color activeButtonTextColor()
             // Handle other system colors below, that don't need special system appearance handling.
             break;
         }
-    } else if (forVisitedLink && cssValueID == CSSValueWebkitLink) {
-        // The system color cache below can't handle visited links. The only color value
-        // that cares about visited links is CSSValueWebkitLink, so handle it here.
+    } else if (forVisitedLink && (cssValueID == CSSValueWebkitLink || cssValueID == CSSValueVisitedtext)) {
```

Comment:
Is the check for `CSSValueVisitedtext` really necessary here? Its value is not actually dependent on `forVisitedLink`, so I would have expected that it's cacheable.

Reply:
It doesn't look necessary from what I can understand

Reply:
Without it the assertion after it fails because the WebKit link is reliant on this value under the hood now.

Reply:
Might be better to change the assertion below so that CSSValueVisitedtext can still be cached:

```
// -webkit-link requests Visitedtext when the link is visited, let it fetch Visitedtext from the cache if possible.
ASSERT(!forVisitedLink || cssValueID == CSSValueVisitedtext);
```

(Although right now Linktext & VisitedText are never cached due to not using a system color, there is a bug to change that)

Reply:
Maybe @pxlcoder has a better idea?

Reply:
I've gone ahead with that change. Seems as reasonable as any other approach.

Reply:
I would have probably factored out the code in `RenderTheme` into static methods, rather than making `-webkit-link` explicitly dependent on `visitedtext` – but this seems fine too.